### PR TITLE
Restructure dev box

### PR DESCRIPTION
**Pull request by luchaos**
Remove dedicated game rename page and integrate into game dev box.
Remove unused redirector comments.
Unify comments layout.

### CODE REVIEW

File: public/gameInfo.php [hunk 1]
```diff
@@ -666,78 +671,74 @@ function ResetProgress() {
             if (isset($user) && ($permissions >= Permissions::Developer || ($isFullyFeaturedGame && $permissions >= Permissions::JuniorDeveloper))) {
                 echo "<div class='devbox mb-3'>";
                 echo "<span onclick=\"$('#devboxcontent').toggle(); return false;\">Dev ▼</span>";
-                echo "<div id='devboxcontent' style='display: none'>";
-
-                // Only allow developers to rename a game
-                if ($permissions >= Permissions::Developer) {
-                    echo "<div class='mb-3'><a href='/attemptrename.php?g=$gameID'>Rename Game</a></div>";
-                }
+                // echo "<div id='devboxcontent' style='display: none'>";
```
Comment (Jamiras) [inline public/gameInfo.php:674]: this `display:none` is required to have the devbox start in a collapsed state. I'm assuming it was commented out for testing and should be restored while removing the next line.

File: public/gameInfo.php [hunk 2]
```diff
@@ -666,78 +671,74 @@ function ResetProgress() {
...
+                    echo "<div><a class='btn btn-link' href='/ticketmanager.php?g=$gameID'>Tickets</a></div>";
+                    echo "<div>";
                     $isSubscribedToTickets = isUserSubscribedTo(SubscriptionSubjectType::GameTickets, $gameID, $userID);
                     RenderUpdateSubscriptionForm(
                         "updateticketssub",
                         SubscriptionSubjectType::GameTickets,
                         $gameID,
                         $isSubscribedToTickets,
-                        'tickets'
+                        'Tickets'
```
Comment (Jamiras) [inline public/gameInfo.php:731]: There's also a "Subscribe to achievement comments" button below the rich presence. It should probably be brought up here.

**Review by Jamiras — COMMENTED** [review]
Overall, I like the changes. I've offered a few additional suggestions since we're already changing things.

File: public/gameInfo.php (hunk 2 above)
Comment (luchaos) [reply public/gameInfo.php:731]: I think I put that down there on purpose to match other comments areas where the subscription button is on top. I can move it back though if you think it feels more natural to have it up there again

File: public/gameInfo.php (hunk 1 above)
Comment (luchaos) [reply public/gameInfo.php:674]: Whoops, that wasn't supposed to be in there indeed. Got confused with my stashes there.

File: public/gameInfo.php [hunk 3]
```diff
@@ -827,54 +833,81 @@ function ResetProgress() {
...
-                    echo "<tr><td>First Released:</td><td><input type='text' name='release' value='" . attributeEscape($released) . "' style='width:100%;'></td></tr>";
-                    echo "<tr><td></td><td class='text-right'><button class='btn'>Submit</button></td></tr>";
-                    echo "</tbody></table>";
+                    echo "<div class='grid grid-cols-[180px_1fr_100px] gap-1 items-center mb-1'>";
+                    echo "<label>Developer</label><input type='text' name='developer' value='" . attributeEscape($developer) . "' class='w-full'><div></div>";
+                    echo "<label>Publisher</label><input type='text' name='publisher' value='" . attributeEscape($publisher) . "' class='w-full'><div></div>";
+                    echo "<label>Genre</label><input type='text' name='genre' value='" . attributeEscape($genre) . "' class='w-full'><div></div>";
+                    echo "<label>First Released</label><input type='text' name='release' value='" . attributeEscape($released) . "' class='w-full'><div></div>";
+                    echo "</div>";
+                    echo "<div class='text-right'><button class='btn'>Submit</button></div>";
```
[the remark this answers is not in the document]
Comment (luchaos) [reply public/gameInfo.php:878]: Was thinking about submit button on each line as well; didn't like the orphaned button too much either. Let's have that 👍 
Moving fields, too, good idea.

File: public/gameInfo.php [hunk 4]
```diff
@@ -666,78 +671,74 @@ function ResetProgress() {
...
+                        echo "<div><a class='btn btn-link' href='/leaderboardList.php?g=$gameID'>Leaderboards</a></div>";
                     }
                     echo "</div>";
 
-                    echo "<div class='mb-1'><a href='/ticketmanager.php?g=$gameID'>View tickets for this game</a></div>";
-                    echo "<div class='mb-3'>";
+                    echo "<div><a class='btn btn-link' href='/codenotes.php?g=$gameID'>Code Notes</a></div>";
+
+                    if ($permissions >= Permissions::Developer) {
+                        echo "<div class='mb-3'><a class='btn btn-link' href='/managehashes.php?g=$gameID'>Hashes</a></div>";
```
[the remark this answers is not in the document]
Comment (luchaos) [reply public/gameInfo.php:708]: Good call, will rename it back.

File: public/gameInfo.php [hunk 5]
```diff
@@ -827,54 +833,81 @@ function ResetProgress() {
...
+                    }
+
+                    // Only allow developers to rename a game
+                    if ($permissions >= Permissions::Developer) {
+                        echo "<form class='mb-2' method='post' action='/request/game/update-title.php'>";
+                        echo csrf_field();
+                        echo "<input type='hidden' name='game' value='$gameID' />";
+                        echo "<div class='grid grid-cols-[180px_1fr_100px] gap-1 items-center mb-1'>";
+                        echo "<label>Name</label>";
+                        echo "<input type='text' name='title' value='$gameTitle' maxlength='80' class='w-full'>";
```
[the remark this answers is not in the document]
Comment (luchaos) [reply public/gameInfo.php:863]: Oof, thanks! Had it in mind but forgot to do so

File: public/gameInfo.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (luchaos) [reply public/gameInfo.php]: Made it "Manage Leaderboards" while at it to be consistent in that column.

File: public/gameInfo.php (hunk 2 above)
Comment (luchaos) [reply public/gameInfo.php:729]: Moved back up.

File: public/gameInfo.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (luchaos) [reply public/gameInfo.php]: Had to restructure the blocks a bit to match the required permissions.

**Comment by luchaos** [comment]
Now I wonder if the claims block should go back into the left column as well, reuniting it with the other management links.

Screenshot 2022-10-03 at 00 26 52

**Comment by luchaos** [comment]
Reverted an unintended change where the line-height of buttons and form inputs was increased.

This wasn't meant to be introduced before a general typography overhaul.

**Comment by Jamiras** [comment]
> Maybe leaving the claim button in the right column? Actions vs. links.

I like that idea. Note that "Manage Leaderboards" becomes a button if there aren't any.

May even want to split them so "Manage Claims" is on the left (it's only visible to admins) and then move Claim History below the Claim/Drop button

**Comment by luchaos** [comment]
> May even want to split them so "Manage Claims" is on the left (it's only visible to admins) and then move Claim History below the Claim/Drop button

🙌 

Screenshot 2022-10-03 at 01 50 48

> Note that "Manage Leaderboards" becomes a button if there aren't any.

I'd say that's fine for now - something for another time? 😅

**Review by Jamiras — APPROVED** [review]

**Pull request merged**